hal/imx_rt1064_nor.h

```diff
@@ -0,0 +1,252 @@
+
+/* wolfBoot interface for iMX-RT1060-EVK.
+ *
+ * Description of the NOR configuration interface required by the board.
+ *
+ * Copyright (C) 2021 wolfSSL Inc.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifndef IMXRT_NOR
+#define IMXRT_NOR
+/* FLEXSPI memory config block related defintions */
+#define FLEXSPI_CFG_BLK_TAG     (0x42464346UL) // ascii "FCFB" Big Endian
+#define FLEXSPI_CFG_BLK_VERSION (0x56010400UL) // V1.4.0
+#define FLEXSPI_CFG_BLK_SIZE    (512)
+
+/* FLEXSPI Feature related definitions */
+#define FLEXSPI_FEATURE_HAS_PARALLEL_MODE 1
+
+/* Lookup table related defintions */
+#define CMD_INDEX_READ        0
+#define CMD_INDEX_READSTATUS  1
+#define CMD_INDEX_WRITEENABLE 2
+#define CMD_INDEX_WRITE       4
+
+#define CMD_LUT_SEQ_IDX_READ        0
+#define CMD_LUT_SEQ_IDX_READSTATUS  1
+#define CMD_LUT_SEQ_IDX_WRITEENABLE 3
+#define CMD_LUT_SEQ_IDX_WRITE       9
+
+#define CMD_SDR        0x01
+#define CMD_DDR        0x21
+#define RADDR_SDR      0x02
+#define RADDR_DDR      0x22
+#define CADDR_SDR      0x03
+#define CADDR_DDR      0x23
+#define MODE1_SDR      0x04
+#define MODE1_DDR      0x24
+#define MODE2_SDR      0x05
+#define MODE2_DDR      0x25
+#define MODE4_SDR      0x06
+#define MODE4_DDR      0x26
+#define MODE8_SDR      0x07
+#define MODE8_DDR      0x27
+#define WRITE_SDR      0x08
+#define WRITE_DDR      0x28
+#define READ_SDR       0x09
+#define READ_DDR       0x29
+#define LEARN_SDR      0x0A
+#define LEARN_DDR      0x2A
+#define DATSZ_SDR      0x0B
+#define DATSZ_DDR      0x2B
+#define DUMMY_SDR      0x0C
+#define DUMMY_DDR      0x2C
+#define DUMMY_RWDS_SDR 0x0D
+#define DUMMY_RWDS_DDR 0x2D
+#define JMP_ON_CS      0x1F
+#define STOP           0
+
+#define FLEXSPI_1PAD 0
+#define FLEXSPI_2PAD 1
+#define FLEXSPI_4PAD 2
+#define FLEXSPI_8PAD 3
+
+#define FLEXSPI_LUT_SEQ(cmd0, pad0, op0, cmd1, pad1, op1)                                                              \
+    (FLEXSPI_LUT_OPERAND0(op0) | FLEXSPI_LUT_NUM_PADS0(pad0) | FLEXSPI_LUT_OPCODE0(cmd0) | FLEXSPI_LUT_OPERAND1(op1) | \
+     FLEXSPI_LUT_NUM_PADS1(pad1) | FLEXSPI_LUT_OPCODE1(cmd1))
+
+//!@brief Definitions for FlexSPI Serial Clock Frequency
```

Comment:
Please use C-style comments only (`/*`)

Reply:
throughout the file

Reply:
Done.

This does bring up one issue I should mention: since that file is almost entirely driver code pulled from the MCUXpresso SDK (`boards/{board}/xip/{board}_flexspi_nor_config.h`), changing all of the comments (or any other similar change) will result in a very messy file diff if you wish to upgrade to a newer SDK version in the future and check what changed. 

The same situation exists right now with the 1050 and 1060 ports in their respective flexpi NOR configuration files (`imx_rt10{50,60}_nor.h`), which is why I did it the way I did from the start. **If adhering to a particular comment style is required, we might want to revisit those files and change the style too?** Happy to do that as part of this PR or introduce it in its own PR.

The real question is whether these NOR configuration files should even be part of wolfBoot source code, or instead just referenced as an external SDK file. If we want to treat it as vendor code/external, we should get rid of the `hal/imx_rtXXXX_nor.h` files in their entirety (since they are just SDK code anyway) and instead have `hal/imx_rt.c` directly `#include "{board}_flexspi_nor_config.h"` from the SDK, reducing copy-pasted code. 

We can always introduce the refactor mentioned above in a later PR, since it would increase the scope and require retesting on all supported platforms. Just food for thought for future improvement.

Reply:
Hi @bigbrett , there is also the issue or intellectual property. If you can reference the NXP SDK for these headers directly please do. If not then these should be re-written. Ping me directly if you'd like to discuss.

Reply:
The files in question do not generate any code, and are only interfaces to the NXP flash drivers API.

I agree however that these files should be referenced externally when linking to the NXP SDK rather than be part of the code distributed in the wolfBoot package.

@bigbrett feel free to remove these for 1050 and 1060 as well, and ensure that they can be found when building with the given configurations according to the include path.

Reply:
@danielinux @dgarske sounds like we are all in agreement then. I just pushed a commit removing the file entirely for all three targets (1050, 1060, and 1064). All three targets build without issues but am only able to test the 1064 port on hardware.